console/console-init/ui/src/modules/msg-and-iot/dailogs/components/IoTProjectConfiguration/IoTProjectConfiguration.tsx

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2020, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+import React, { useState } from "react";
+import { isIoTProjectValid } from "modules/msg-and-iot/dailogs/utils";
+import { dnsSubDomainRfc1123NameRegexp } from "utils";
+import { IDropdownOption } from "components";
+import { IoTConfiguration } from "../IoTConfiguration";
+
+export interface IIoTProjectInput {
+  iotProjectName?: string;
+  namespace?: string;
+  isEnabled?: boolean;
+  isNameValid?: boolean;
+}
+
+const IoTProjectConfiguration = (
+  setProjectDetail: (value: IIoTProjectInput) => void,
+  namespaces: IDropdownOption[],
+  projectDetail?: IIoTProjectInput
+) => {
+  const [iotProjectName, setIotProjectName] = useState<string>(
+    (projectDetail && projectDetail.iotProjectName) || ""
+  );
+  const [iotProjectNamespace, setIotProjectNamespace] = useState<string>(
+    (projectDetail && projectDetail.namespace) || ""
+  );
+  const [isIotEnabled, setIsIotEnabled] = useState<boolean>(true);
+  const [isNameValid, setIsNameValid] = useState<boolean>(true);
+  const isReviewEnabled = () => {
+    isIoTProjectValid(projectDetail);
+  };
+  const setData = (value?: string, type?: string, booleanValue?: boolean) => {
+    let projectCopy: IIoTProjectInput = projectDetail || {};
```

Comment:
why are we creating copy of project?

Reply:
Updated the logic and removed all the state variable as seems to be useless